src/react/components/shell/mainContentRouter.tsx

```diff
@@ -12,16 +12,19 @@ import ProfileSettingsPage from "../pages/profileSettingsPage";
 export default function MainContentRouter() {
     return (
         <div className="app-content">
-            <Route path="/" exact component={HomePage} />
-            <Route path="/settings" component={AppSettingsPage} />
-            <Route path="/profile" component={ProfileSettingsPage} />
-            <Route path="/connections/:connectionId" component={ConnectionPage} />
-            <Route path="/connections" exact component={ConnectionPage} />
-            <Route path="/projects/:projectId/edit" component={EditorPage} />
-            <Route path="/projects/create" component={ProjectSettingsPage} />
-            <Route path="/projects/:projectId/settings" component={ProjectSettingsPage} />
-            <Route path="/projects/:projectId/export" component={ExportPage} />
-            <Route path="/projects/:projectId/active-learning" component={ActiveLearningPage} />
+            <Switch>
+                <Route path="/" exact component={HomePage} />
+                <Route path="/settings" component={AppSettingsPage} />
+                <Route path="/profile" component={ProfileSettingsPage} />
+                <Route path="/connections/:connectionId" component={ConnectionPage} />
+                <Route path="/connections" exact component={ConnectionPage} />
+                <Route path="/projects/:projectId/edit" component={EditorPage} />
+                <Route path="/projects/create" component={ProjectSettingsPage} />
+                <Route path="/projects/:projectId/settings" component={ProjectSettingsPage} />
+                <Route path="/projects/:projectId/export" component={ExportPage} />
+                <Route path="/projects/:projectId/active-learning" component={ActiveLearningPage} />
+                <Route component={HomePage} />
```

Comment:
do we need a `path="*"` attribute?  Referenced from https://stackoverflow.com/questions/32128978/react-router-no-not-found-route

Reply:
instead of "*", I wrapped the router in a Switch, which allows fallthrough to the final, default route.